### PR TITLE
LINKTA-170: Update MainLayout structure to work with React Router

### PR DESCRIPTION
**Pull request by kinskiwu**
## Description

- Removed the showFooter prop from the `MainLayout` component.
- Integrated dynamic footer visibility logic into the `useDynamicNavigation` hook.
- Introduced `routeToFooterVisibilityMapping` in `layoutConfig` to control footer display based on the current route.
- Increase TypeScript coverage 

Fixes #LINKTA-170
Related PR: #46 

## Type of change

- [X] Other 🗒️ refactor component to work with current routing

## How Has This Been Tested?

- [ ] Manual testing by running npm run dev

## Checklist:

- [X] I have performed a self-review of my own code.
- [X] I have commented on my code, particularly in hard-to-understand areas.
- [X] My changes generate no new warnings.
- [X] Any dependent changes have been merged and published in downstream modules.
- [X] I have reviewed the [**Security Best Practices Document**](SECURITY_BEST_PRACTICES.md).
- [X] I have followed the [**Naming Conventions Guide**](NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [X] I have ensured that my pull request title is descriptive.

### CODE REVIEW

File: client/components/layout/main-layout/MainLayout.tsx [hunk 1]
```diff
@@ -1,24 +1,28 @@
 import React from 'react';
 import { Outlet } from 'react-router-dom';
-import Header from './header/Header';
 import Footer from './footer/Footer';
```
Comment (philzmintmojito) [inline client/components/layout/main-layout/MainLayout.tsx:3]: This can be removed as the Footer is not used in the current version of main layout.

**Review by philzmintmojito — APPROVED** [review]
LGTM, minor change left in comments.
image

File: client/components/layout/main-layout/MainLayout.tsx (hunk 1 above)
Comment (kinskiwu) [reply client/components/layout/main-layout/MainLayout.tsx:3]: Do you mean the Footer component? The Footer is part of the structure for the MainLayout. It's totally fine to keep it there, even if it's still a work in progress.

**Review by ejwoods — COMMENTED** [review]
Ready to approve pending #46 merge

**Review by ejwoods — APPROVED** [review]

**Pull request merged**